bertopic/_utils.py

```diff
@@ -226,3 +226,28 @@ def to_ndarray(array: Union[np.ndarray, csr_matrix]) -> np.ndarray:
             repr_, ctfidf_used = embeddings, False
 
     return to_ndarray(repr_) if output_ndarray else repr_, ctfidf_used
+
+
+# Visualization mocks in case plotly is not installed
+class MockPlotting:
+    """Mock plotting module when plotly is not installed."""
+
+    def __init__(self, logger: MyLogger):
+        self.logger = logger
+
+    def __getattr__(self, name):
+        def mock_function(*args, **kwargs):
+            self.logger.warning(f"Plotly is not installed. Cannot use {name} visualization function.")
+            return MockFigure()
+
+        return mock_function
+
+
+class MockFigure:
+    """Mock class for plotly.graph_objects.Figure when plotly is not installed."""
+
+    def __init__(self, *args, **kwargs):
+        pass
```

Comment:
I don't think this is necessary when we can use `def some_function(self) -> "go.Figure"` instead. This would reduce the code quite a bit.